docs: improve updating pre-commit documentation (#1361)
copier-org/copier: docs/updating.md

## Preventing Commit of Merge Conflicts

> If you use `--conflict inline` (the default) then you need to check for conflicts marks in your files: 

Reviewer: * `marks` ->  `markers`
* An empty line is missing.
* The line length exceeds 88 characters, so [Prettier won't be happy](https://github.com/copier-org/copier/blob/5fa3b9fb772101ac67476ab235b8e66ca26900de/.prettierrc.yml#L1).

```suggestion
If you use `--conflict inline` (the default) then you need to check for conflicts
markers in your files:

```

```yaml title=".pre-commit-config.yaml"
repos:
    - repo: https://github.com/pre-commit/pre-commit-hooks
      rev: v4.3.0
      hooks:
          # Prevent committing inline conflict markers
          - id: check-merge-conflict
            args: [--assume-in-merge]
```

> If you use `--conflict rej` Then you need to reject all the generated `.rej` files:

Reviewer: Some spelling, wording and content improvements:

```suggestion
If you use `--conflict rej` then you need to review and remove all generated `.rej`
files:
```

          # Prevent committing .rej files
          - id: forbidden-files
            name: forbidden files

>             entry: found Copier update rejection files; review and remove them before merging.

Reviewer: This was actually incorrect before: [`entry`](https://pre-commit.com/#hooks-entry) refers to the executable to run. This is a [description](https://pre-commit.com/#hooks-description):

```suggestion
            description: found Copier update rejection files; review and remove them before merging.
```

            language: fail
            files: "\\.rej$"
```

!!! note

>     For shared project that use both "rej" and "inline" depending on the user,
>     you can add both repo to you `pre-commit-config.yaml` file, making sure that none
>     of these files are committed.

Reviewer: Just some minor suggestions for better clarity and grammatical correctness:

```suggestion
    For projects that use both `rej` and `inline` depending on each user's preference,
    you can add both hooks to your `pre-commit-config.yaml` file, making sure that no
    unresolved merge conflicts are committed.
```
